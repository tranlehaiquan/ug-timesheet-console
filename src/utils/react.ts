import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
