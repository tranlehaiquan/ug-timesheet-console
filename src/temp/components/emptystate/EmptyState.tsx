import * as React from "react";
import classnames from "classnames";

interface IEmptyState extends React.HTMLAttributes<HTMLDivElement> {
  imgSrc: string;
  title: string;
  message: string;
  className?: string;
}

export const EmptyState: React.SFC<IEmptyState> = ({
  imgSrc,
  title,
  message,
  className,
  children,
  ...otherProps
}) => {
  return (
    <div
      className={classnames("sk-text-center sk-leading-normal", className)}
      {...otherProps}
    >
      <img src={imgSrc} className="sk-w-24" />
      <div className="sk-pt-5 sk-font-medium sk-text-base sk-text-navy-lighter">
        {title}
      </div>
      <div className="sk-text-navy-lighter sk-font-normal sk-text-sm sk-mb-4">
        {message}
      </div>
      {children}
    </div>
  );
};
