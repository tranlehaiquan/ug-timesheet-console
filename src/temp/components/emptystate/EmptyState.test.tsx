import * as React from "react";
import { render } from "enzyme";

import { EmptyState } from "./EmptyState";

describe("EmptyState", () => {
  test("render", () => {
    const wrapper = render(
      <EmptyState
        imgSrc="test.png"
        title="Empty state title"
        message="Sub title"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
