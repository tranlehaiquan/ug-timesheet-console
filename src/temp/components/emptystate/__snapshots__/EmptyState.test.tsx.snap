// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState render 1`] = `
<div
  class="sk-text-center sk-leading-normal"
>
  <img
    class="sk-w-24"
    src="test.png"
  />
  <div
    class="sk-pt-5 sk-font-medium sk-text-base sk-text-navy-lighter"
  >
    Empty state title
  </div>
  <div
    class="sk-text-navy-lighter sk-font-normal sk-text-sm sk-mb-4"
  >
    Sub title
  </div>
</div>
`;
